### ISSUE TITLE
ESM error when trying to use with Next.js

### ISSUE THREAD

**Issue by ChristianIvicevic** [apollographql/embeddable-explorer#160]
I tried creating a Next.js page to embed a local Sandbox instance since I cannot use the cloud-based instance to connect to my local API. However there are unexpected issues when doing so.

Error Message

```
Error [ERR_REQUIRE_ESM]: require() of ES Module node_modules/@apollo/sandbox/index.development.js from node_modules/@apollo/sandbox/index.cjs not supported.
index.development.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename index.development.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in node_modules/@apollo/sandbox/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).
```

Reproduction:

1. Checkout https://github.com/ChristianIvicevic/apollo-studio-esm-reproduction
2. Run `npm run dev`
3. Navigate to `localhost:3000/sandbox`

**Comment by brittmmendez** [comment]
We are also getting the same issue... Any tips on how to fix this on our end or is this something apollographql will have to update?

**Comment by ChristianIvicevic** [comment]
> We are also getting the same issue... Any tips on how to fix this on our end or is this something apollographql will have to update?

Using the `embed` option on the server within the API router itself (without using the the explorer package) I was able to circumvent the issue. Here is my entire API route that works for me:
```ts
import {
  ApolloServerPluginLandingPageLocalDefault,
  ApolloServerPluginLandingPageProductionDefault
} from 'apollo-server-core';
import { ApolloServer } from 'apollo-server-micro';
import cors from 'micro-cors';
import type { NextApiRequest, NextApiResponse, PageConfig } from 'next';
import { createGraphQLContext } from 'server/context';
import { graphQLSchema } from 'server/schema';

const apolloServer = new ApolloServer({
  schema: graphQLSchema,
  context: createGraphQLContext,
  plugins: [
    process.env.NODE_ENV === 'production'
      ? ApolloServerPluginLandingPageProductionDefault({ footer: false })
      : ApolloServerPluginLandingPageLocalDefault({ footer: false, embed: true })
  ]
});
const server = apolloServer.start();

const handler = cors({ origin: 'https://studio.apollographql.com' })(
  // @ts-expect-error
  async (req: NextApiRequest, res: NextApiResponse) => {
    await server;
    const handler = apolloServer.createHandler({
      path: '/api/graphql'
    });
    if (req.method === 'OPTIONS') {
      res.status(200).send('OK');
    } else {
      return handler(req, res);
    }
  }
);

export default handler;

export const config: PageConfig = {
  api: {
    bodyParser: false
  }
};
```

**Comment by steventhan** [comment]
I ran into the same issue, my workaround is to use the CDN-hosted version:

```tsx
import Script from "next/script";

export default function EmbeddedSandbox() {
  return (
    <>
      <div styles={{ height: "100vh" }} id="embedded-sandbox"></div>)
      <Script
        src="https://embeddable-sandbox.cdn.apollographql.com/_latest/embeddable-sandbox.umd.production.min.js"
        onReady={() => {
          new (window as any).EmbeddedSandbox({
            target: "#embedded-sandbox",
            initialEndpoint: "http://localhost:4000/graphql",
            includeCookies: false,
          });
        }}
      />
    </>
  );
}
```

**Comment by mayakoneval** [comment]
Hi! Thanks for reporting this, I have made a ticket on our board to fix this & I am able to reproduce.

**Comment by snowiewdev** [comment]
> I ran into the same issue, my workaround is to use the CDN-hosted version:
> 
> ```tsx
> import Script from "next/script";
> 
> export default function EmbeddedSandbox() {
>   return (
>     <>
>       <div styles={{ height: "100vh" }} id="embedded-sandbox"></div>)
>       <Script
>         src="https://embeddable-sandbox.cdn.apollographql.com/_latest/embeddable-sandbox.umd.production.min.js"
>         onReady={() => {
>           new (window as any).EmbeddedSandbox({
>             target: "#embedded-sandbox",
>             initialEndpoint: "http://localhost:4000/graphql",
>             includeCookies: false,
>           });
>         }}
>       />
>     </>
>   );
> }
> ```

Thanks for the workaround. Do you have warnings about CORS in the console? If so, how did you resolve it? 

I use the CDN approach but the console has multiple warnings regarding on CORS. I ran ```npx diagnose-endpoint@1.1.0 --endpoint=MY_LOCAL_GRAPHQL_ENDPOINT``` and it returns messages:
⚠️  OPTIONS response is missing header 'access-control-allow-methods: POST'
⚠️  POST response missing 'access-control-allow-origin' header.

**Comment by mayakoneval** [comment]
Hello, this should be fixed in @apollo/explorer@2.0.1 and @apollo/sandbox@1.0.2 now!
